src/mpid/ch4/shm/posix/posix_am.h

```diff
@@ -83,6 +83,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_am_enqueue_request(const void *am_hdr,
 
 MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_am_isend(int rank,
                                                   MPIR_Comm * comm,
+                                                  MPIDI_POSIX_am_header_kind_t kind,
```

Comment:
I think you are right. XPMEM needs to be based on POSIX function in this case (sending a header to tell receiver exposed data address). This code is written by Min, and she may not want to introduce the duplicate code; that's why she chose to change the interface. Should we also discuss this part with Min if possible? Just want to make sure what she thinks. I think we can have a `MPIDI_XPMEM_am_send_hdr` and design its interface on our own, but basically will call `MPIDI_POSIX_am_isend`.

Reply:
@gcongiu Sorry for the delayed response. We will need SHM internal AM also in other shmmods (e.g., CMA, PIP). It will be terrible to duplicate the send code everywhere. I am not sure if I understood your concern "leak transport specific information to the above layers ." POSIX is the basic transport infrastructure for all other shmmods, and it is directly called only by the SHM glue layer. SHM glue layer should define whether the packet is from CH4 or an internal AM, so I think it is valid to let the glue layer pass `kind` to POSIX similar to existing `handler_id` parameter.

Reply:
@minsii sorry for the lack of clarity, you can disregard that comment as I was basically suggesting to add the `kind` field to the AM header but also making a note about how the AM should not probably contain such information as the AM code does not need to be concerned with underlying mechanism used for data transfer.

Reply:
I do not think the `kind` field can be added into the am header. The am header is defined by either CH4 (for generic AM) or SHM (for SHM internal control message), and its internal fields should not be exposed to the POSIX transport layer. However, the POSIX transport needs to know whether it is a CH4 generic message or a SHM internal control message in order to call the corresponding callback. Therefore, the `kind` field has to be managed by the transport layer.